tests/foreman/ui/test_role.py

```diff
@@ -99,14 +102,127 @@ def test_positive_update_permission(self):
+        with Session(self.browser):
+            self.role.clone(builtin_name, new_name)
+            self.assertIsNotNone(
+                self.role.wait_until_element(common_locators['alert.success']))
+            self.assertIsNotNone(self.role.search(new_name))
+            # Ensure that cloned role contains correct resource type
+            builtin_resources = self.role.get_resources(builtin_name)
+            cloned_resources = self.role.get_resources(new_name)
+            self.assertGreater(len(cloned_resources), 0)
+            self.assertEqual(set(builtin_resources), set(cloned_resources))
+            # And correct permissions for every resource type
+            for resource in cloned_resources:
+                self.assertEqual(
+                    set(self.role.get_permissions(builtin_name, resource)),
+                    set(self.role.get_permissions(new_name, resource))
+                )
+
+    @tier1
+    def test_positive_clone_custom(self):
+        """Create custom role with permissions and clone it
+
+        @id: a4367968-eae5-4b8a-9b5c-61824b261320
+
+        @Assert: New role is created and contains all permissions
+        """
+        name = gen_string('alpha')
+        new_name = gen_string('alpha')
+        # Pick up custom permissions
+        resource_type = 'Architecture'
+        permissions = ['view_architectures', 'edit_architectures']
+        with Session(self.browser) as session:
+            # Create custom role with permissions
+            make_role(session, name=name)
+            self.role.add_permission(
+                name, resource_type=resource_type, permission_list=permissions,
             )
+            self.assertIsNotNone(self.role.search(name))
+            # Clone role
+            self.role.clone(name, new_name)
+            self.assertIsNotNone(
+                self.role.wait_until_element(common_locators['alert.success']))
+            self.assertIsNotNone(self.role.search(new_name))
+            # Ensure that cloned role contains correct resource type
+            cloned_resources = self.role.get_resources(new_name)
+            self.assertGreater(len(cloned_resources), 0)
+            self.assertEqual(resource_type, cloned_resources[0])
+            # and all permissions
+            cloned_permissions = self.role.get_permissions(
+                new_name, resource_type)
+            self.assertIsNotNone(cloned_permissions)
+            self.assertEqual(set(permissions), set(cloned_permissions))
+
+    @tier1
+    def test_positive_assign_cloned_role(self):
+        """Clone role and assign it to user
+
+        @id: cbb17f37-9039-4875-981b-1f427b095ed1
+
+        @Assert: User is created successfully
+
+        @BZ: 1353788
+        """
+        user_name = gen_string('alpha')
+        role_name = gen_string('alpha')
+        with Session(self.browser) as session:
+            # Clone one of the builtin roles
+            self.role.navigate_to_entity()
```

Comment:
At first I tried to implement this test within `test_user` suite as role can be assigned only from user page, but there was a problem with navigation to `Roles` page, so apparently it doesn't.